curl/inc/leatherman/curl/client.hpp

```diff
@@ -110,31 +146,50 @@ namespace leatherman { namespace curl {
     };
 
     /**
-     * The exception for HTTP file downloads.
+     * The exception for curl_easy_setopt errors.
      */
-    struct LEATHERMAN_CURL_EXPORT http_file_download_exception : http_request_exception
+    struct LEATHERMAN_CURL_EXPORT http_setup_exception : http_request_exception
     {
         /**
-         * Constructs an http_file_download_exception.
-         * @param request The request that caused the exception
-         * @param file_path The file that was meant to be downloaded
+         * Constructs an http_setup_exception.
+         * @param req The HTTP request that caused the exception.
          * @param message The exception message.
+         * @param curl_opt The CURL option that failed.
          */
-        http_file_download_exception(request req, std::string file_path, std::string const &message) : http_file_download_exception(req, file_path, "", message)
+        http_setup_exception(request req, CURLoption curl_opt, std::string const &message) :
+            http_request_exception(req, message),
+            _curl_opt(std::move(curl_opt))
         {
         }
 
         /**
-         * Constructs an http_file_download_exception.
+         * Gets the CURL option associated with the exception
+         * @return Returns the CURL option associated with the exception.
+         */
+        CURLoption const& curl_opt() const
+        {
+            return _curl_opt;
+        }
+
+
+     private:
+        CURLoption _curl_opt;
+    };
+
+    /**
+     * The exception for HTTP file download server-side errors.
+     */
+    struct LEATHERMAN_CURL_EXPORT http_download_exception : http_request_exception
```

Comment:
@ekinanp We could've avoided making this a breaking change by keeping the name http_file_download_exception. I'm tempted to do that.